x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/TokenAuthIntegTests.java

```diff
@@ -387,8 +386,11 @@ public void testRefreshingMultipleTimesWithinWindowSucceeds() throws Exception {
         }
         completedLatch.await();
         assertThat(failed.get(), equalTo(false));
-        // Assert that we only ever got one access_token/refresh_token pair
-        assertThat(tokens.stream().distinct().collect(Collectors.toList()).size(), equalTo(1));
+        // Assert that we only ever got one anot(contccess_token/refresh_token pair
+        assertThat((int) tokens.stream().distinct().count(), equalTo(1));
+        // Assert that all requests from all threads could authenticate at the time they received the access token
+        // see: https://github.com/elastic/elasticsearch/issues/54289
+        assertThat(authStatuses, not(hasItem(RestStatus.UNAUTHORIZED)));
```

Comment:
nits: alternatively, maybe we could verify all statuses are `OK` since `getAuthenticationResponseCode` always returns `OK` when no exception is thrown. It feels more precise.

Reply:
I was thinking in terms of testing for the bug this is solving but these all should be OK so I see your point, will address